controllers/volumegroup_controller.go

```diff
@@ -150,17 +150,25 @@ func (r *VolumeGroupReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	if err = r.removeVolumesFromVG(logger, instance); err != nil {
-		return ctrl.Result{}, utils.HandleErrorMessage(logger, r.Client, instance, err, removingPVC)
-	}
-	if err = r.addMatchingVolumesToVG(logger, instance); err != nil {
-		return ctrl.Result{}, utils.HandleErrorMessage(logger, r.Client, instance, err, addingPVC)
+	err = r.updatePVCs(err, logger, instance)
+	if err != nil {
+		return ctrl.Result{}, err
 	}
 
 	r.createSuccessVolumeGroupEvent(logger, instance)
 	return ctrl.Result{}, utils.HandleErrorMessage(logger, r.Client, instance, err, vgReconcile)
 }
 
+func (r *VolumeGroupReconciler) updatePVCs(err error, logger logr.Logger, instance *volumegroupv1.VolumeGroup) error {
+	if err = r.removeVolumesFromVG(logger, instance); err != nil {
+		return utils.HandleErrorMessage(logger, r.Client, instance, err, removingPVC)
+	}
+	if err = r.addMatchingVolumesToVG(logger, instance); err != nil {
+		return utils.HandleErrorMessage(logger, r.Client, instance, err, addingPVC)
```

Comment:
IMO we should handle the `utils.HandleErrorMessage` function in the reconcile function as we do with the other functions (can be done in a separate ticket because of GA)